RecoLocalCalo/HcalRecAlgos/src/MahiFit.cc

```diff
@@ -212,9 +215,10 @@ const float MahiFit::minimize(const double noisecorr) const {
   invCovMat += nnlsWork_.noiseTerms.asDiagonal();
 
   //Add off-Diagonal components up to first order
-  if (noisecorr != 0) {
+  if (nnlsWork_.noisecorr != 0) {
     for (unsigned int i = 1; i < nnlsWork_.tsSize; ++i) {
-      auto const noiseCorrTerm = noisecorr * sqrt(nnlsWork_.noiseTerms.coeff(i - 1) * nnlsWork_.noiseTerms.coeff(i));
+      auto const noiseCorrTerm =
+          nnlsWork_.noisecorr * sqrt(nnlsWork_.pedVals.coeff(i - 1) * nnlsWork_.pedVals.coeff(i));
```

Comment:
```suggestion
          nnlsWork_.noisecorr * nnlsWork_.pedVals.coeff(i - 1) * nnlsWork_.pedVals.coeff(i);
```
to match the suggested change above

Reply:
Thanks for the suggestion!